lib/icinga/downtime.cpp

```diff
@@ -369,19 +369,19 @@ void Downtime::RemoveDowntime(const String& id, bool includeChildren, bool cance
 
 	String config_owner = downtime->GetConfigOwner();
 
-	if (!config_owner.IsEmpty() && !expired) {
+	if (!config_owner.IsEmpty() && removalReason == DowntimeRemovedByUser) {
 		BOOST_THROW_EXCEPTION(invalid_downtime_removal_error("Cannot remove downtime '" + downtime->GetName() +
 			"'. It is owned by scheduled downtime object '" + config_owner + "'"));
 	}
 
 	if (includeChildren) {
 		for (const Downtime::Ptr& child : downtime->GetChildren()) {
-			Downtime::RemoveDowntime(child->GetName(), true, true);
+			Downtime::RemoveDowntime(child->GetName(), true, removalReason, removedBy);
 		}
 	}
 
-	if (cancelled) {
-		downtime->SetRemovalInfo(removedBy, Utility::GetTime());
+	if (removalReason != DowntimeExpired) {
+		downtime->SetRemovalInfo(removedBy, Utility::GetTime(), origin);
```

Comment:
Why don't we omit ALL the origin passthrough additions for now and discuss them separately?

Reply:
No problem! `Downtime::RemoveDowntime()` is never called from outside with the `origin` param set anyway, so I am wondering why it was introduced in the first place.